gcc/rust/backend/rust-mangle.cc

```diff
@@ -79,6 +79,92 @@ legacy_mangle_self (const TyTy::BaseType *self)
   return legacy_mangle_name (buf);
 }
 
+static std::string
+v0_tuple_prefix (const TyTy::BaseType *ty)
+{
+  if (ty->is_unit ())
+    return "u";
+
+  // FIXME: ARTHUR: Add rest of algorithm
+  return "";
+}
+
+static std::string
+v0_numeric_prefix (const TyTy::BaseType *ty)
```

Comment:
I think in later PR's you might want to make this more error safe by doing something like:

```
static bool v0_numeric_prefix(const TyTy::BaseType* ty, std::string& prefix)
```

So in future changes you can error by returning false, so it becomes:

```
std::string prefix;
bool ok = v0_numeric_prefix(ty, prefix);
rust_asser(ok);
```